src/sidebar/test/media-embedder-test.js

```diff
@@ -90,6 +90,57 @@ describe('media-embedder', function () {
     });
   });
 
+  it('replaces mp3 links with html5 audio elements', function() {
+    var urls = [
+      'https://archive.org/download/testmp3testfile/mpthreetest.mp3',
+      'https://archive.org/download/testmp3testfile/mpthreetest.mp3?foo=bar&id=1',
+    ];
+    urls.forEach(function (url) {
+      var element = domElement('<a href="' + url + '">' + url + '</a>');
+
+      mediaEmbedder.replaceLinksWithEmbeds(element);
+
+      assert.equal(element.childElementCount, 1);
+      assert.equal(element.children[0].tagName, 'AUDIO');
+      assert.equal(
+        element.children[0].src, 'https://archive.org/download/testmp3testfile/mpthreetest.mp3'); //not sure if this is correct
```

Comment:
The "src" here will always be equal to the input `url`, given how you implemented the `audioElement` function.